File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> food) {
    string answer = "";
    
    string ans = "";
    for(int i = 1; i < food.size(); i++) {
        int cnt = food[i] / 2;
        for(int j = 0; j < cnt; j++) {
            ans += to_string(i);
        }
    }
        
    answer = ans + "0";
    reverse(ans.begin(), ans.end());
    answer += ans;
    
    return answer;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cs
using System;
using System.Linq;

public class Solution {
    public string solution(int[] food) {
        string answer = "";
        
        for(int i = 1; i < food.Length; i++) {
            int cnt = food[i] / 2;
            for(int j = 0; j < cnt; j++) {
                answer += i.ToString();
            }
        }
        
        string ans_reverse = new string(answer.Reverse().ToArray());
        answer += "0";
        answer += ans_reverse;
        
        return answer;
    }
}